File: julia/src/ModDelayConfig.jl
# File: ModDelayConfig.jl
# Project: src
# Created Date: 14/06/2022
# Author: Shun Suzuki
# -----
# Last Modified: 14/06/2022
# Modified By: Shun Suzuki (suzuki@hapis.k.u-tokyo.ac.jp)
# -----
# Copyright (c) 2022 Shun Suzuki. All rights reserved.
# 


mutable struct ModDelayConfig
    _header_ptr
    function ModDelayConfig()
        chandle = Ref(Ptr{Cvoid}(0))
        autd3capi.autd_create_mod_delay_config(chandle)
        c = new(chandle[])
        finalizer(c -> autd3capi.autd_delete_mod_delay_config(c._header_ptr), c)
        c
    end
end

File: matlab/autd3/ModDelayConfig.m
%{
%File: ModDelayConfig.m
%Project: autd3
%Created Date: 10/06/2022
%Author: Shun Suzuki
%-----
%Last Modified: 10/06/2022
%Modified By: Shun Suzuki (suzuki@hapis.k.u-tokyo.ac.jp)
%-----
%Copyright (c) 2022 Shun Suzuki. All rights reserved.
%
%}

classdef ModDelayConfig < Header

    methods

        function obj = ModDelayConfig()
            obj = obj@Header();
            pp = libpointer('voidPtrPtr', obj.ptr);
            calllib('autd3capi', 'AUTDCreateModDelayConfig', pp);
        end

        function delete(obj)

            if obj.ptr.Value ~= 0
                calllib('autd3capi', 'AUTDDeleteModDelayConfig', obj.ptr);
                obj.ptr = libpointer('voidPtr', 0);
            end

        end

    end

end
